### PR TITLE
fix(attendee registration): Filter tickets shown on attendee registration page to one provider.

### PR DESCRIPTION
**Pull request by Camwyn**
Filter the attendee registration display to only show tickets for the most recent provider. Also
ensure that the checkout button directs to the correct cart. Add provider to attendee registration url.

:ticket: https://central.tri.be/issues/122317

### CODE REVIEW

File: src/Tribe/Attendee_Registration/View.php [hunk 1]
```diff
@@ -30,21 +30,28 @@ public function display_attendee_registration_page( $content = '', $context = 'd
...
 		}
+
+		$q_provider = tribe_get_request_var( 'provider', false );
 		/**
 		 * Filter to add/remove tickets from the global cart
 		 *
 		 * @since TDB
 		 *
-		 * @param array  The array containing the cart elements. Format arrat( 'ticket_id' => 'quantity' );
+		 * @param array  The array containing the cart elements. Format array( 'ticket_id' => 'quantity' );
```
Comment (bordoni) [inline src/Tribe/Attendee_Registration/View.php]: add the second param on the docblock.

File: src/Tribe/Tickets.php [hunk 2]
```diff
@@ -2473,7 +2473,7 @@ public function maybe_redirect_to_attendees_registration_screen( $redirect = nul
 			$storage = new Tribe__Tickets_Plus__Meta__Storage();
 			if ( ! empty( $redirect ) ) {
 				$key = $storage->store_temporary_data( $redirect );
-				$url = add_query_arg( array( 'event_tickets_redirect_to' => $key ), $url );
+				$url = add_query_arg( array( 'event_tickets_redirect_to' => $key, 'provider' => 'tribe_tpp_attendees' ), $url );
```
Comment (bordoni) [inline src/Tribe/Tickets.php]: Dont we have a variable holding this `tribe_tpp_attendees` string?

File: src/Tribe/Attendee_Registration/View.php (hunk 1 above)
Comment (sc0ttkclark) [reply src/Tribe/Attendee_Registration/View.php]:
```suggestion
		 * @param array $cart_tickets The array containing the cart elements. Format array( 'ticket_id' => 'quantity' );
		 * @param string $q_provider  Current ticket provider.
```

File: src/Tribe/Tickets.php (hunk 2 above)
Comment (sc0ttkclark) [reply src/Tribe/Tickets.php]:
```suggestion
				$commerce_paypal = tribe( 'tickets.commerce.paypal' );

				$url = add_query_arg( array( 'event_tickets_redirect_to' => $key, 'provider' => $commerce_paypal->attendee_object ), $url );
```

File: src/Tribe/Attendee_Registration/View.php [hunk 3]
```diff
@@ -30,21 +30,28 @@ public function display_attendee_registration_page( $content = '', $context = 'd
 		if ( 'shortcode' !== $context && ! tribe( 'tickets.attendee_registration' )->is_on_page() ) {
 			return $content;
 		}
+
+		$q_provider = tribe_get_request_var( 'provider', false );
```
Comment (sc0ttkclark) [inline src/Tribe/Attendee_Registration/View.php]:
```suggestion
		$q_provider = tribe_get_request_var( 'provider', false );

```

File: readme.txt [hunk 4]
```diff
@@ -120,6 +120,7 @@ Currently, the following add-ons are available for Event Tickets:
 = [4.10.4] TBD =
 
 * Tweak - After deleting attendees you are now redirected back to the admin attendee page to clear the url of the deleting actions [122083]
+* Fix - Filter the attendee registration display to only show tickets for the current provider. Add provider to attendee registration url. [122317]
```
Comment (sc0ttkclark) [inline readme.txt]:
```suggestion
* Fix - Filter the attendee registration display to only show tickets for the current provider. Add provider to attendee registration url [122317]
```

**Pull request merged**